apps/src/templates/Dialog.jsx

```diff
@@ -138,8 +152,24 @@ const Dialog = React.createClass({
     onCancel: whenNoChildOfTypes(Buttons),
     confirmText: whenNoChildOfTypes(Buttons),
     onConfirm: whenNoChildOfTypes(Buttons),
+    confirmType: whenNoChildOfTypes(Buttons),
   }),
 
+  handleKeyDown(event) {
+    // Always focus the Cancel or Confirm button when tab is pressed, to prevent the
+    // user from selecting elements outside of the dialog.
+    if (event.key === 'Tab') {
+      const cancelButton = this.refs.buttons.getCancelButton();
+      const confirmButton = this.refs.buttons.getConfirmButton();
+      if ((document.activeElement === cancelButton)) {
+        confirmButton && confirmButton.focus();
+      } else {
+        cancelButton && cancelButton.focus();
+      }
+      event.preventDefault();
```

Comment:
yeah, that would be better. I think you might even be able to use instanceof checks instead of css classes to find the cancel button, which IMO would be better.